sound/soc/soc-acpi.c

```diff
@@ -34,7 +34,7 @@ static acpi_status snd_soc_acpi_find_package(acpi_handle handle, u32 level,
 					     void *context, void **ret)
 {
 	struct acpi_device *adev;
-	acpi_status status = AE_OK;
+	acpi_status status;
```

Comment:
Not related to this patch, but should we return `status` in line 65?
```
		if (ACPI_FAILURE(status)) {
			kfree(buffer.pointer);
			return AE_OK;                        -> Should it be return status?
		}
```

Reply:
...and `kfree(buffer.pointer);` could be moved one line up from line 64 and removed from line 68 :-D But yes, unrelated

Reply:
> Not related to this patch, but should we return `status` in line 65?
> 
> ```
> 		if (ACPI_FAILURE(status)) {
> 			kfree(buffer.pointer);
> 			return AE_OK;                        -> Should it be return status?
> 		}
> ```

the AE_OK is actually an error value that will prompt the ACPI code to move to the next element. We always return AE_OK, except for the case where we return AE_TERMINATE which essentially means success.

don't ask me why it was designed this way, this is probably 30 years old now ;-)